File: tp1/ejercicio2/interpretado/decimalBinario.py
# codigo para pasar de binario a decimal y viceversa 

# 1. Definicion de funciones

def binarioADecimal(binario):
    
    decimal = 0
    for i in range(len(binario)):
        decimal += int(binario[i]) * (2 ** (len(binario) - i - 1))
    return decimal

def decimalABinario(decimal):
    
    if decimal == 0:
        return "0"
    binario = ""
    while decimal > 0:
        binario = str(decimal % 2) + binario
        decimal //= 2
    return binario


def main():
        
        # 2. Entrada de datos
        binario = input("Introduce un número binario: ")
        decimal = int(input("Introduce un número decimal: "))
    
        # 3. Proceso y salida de datos
        print(f"El número binario {binario} en decimal es: {binarioADecimal(binario)}")
        print(f"El número decimal {decimal} en binario es: {decimalABinario(decimal)}")
        
if __name__ == "__main__":
    main()

File: tp1/ejercicio2/compilado/decimalBinario.cpp
// hacer un codigo para pasar de decimal a binario y binario a decimal

#include <iostream>
#include <cmath>
#include <string>

using namespace std;

int decimalABinario(int n) {
    int binario = 0, base = 1;
    while (n > 0)
    {
        int ultimoDigito = n % 2;
        n /= 2;
        binario += ultimoDigito * base;
        base *= 10;
    }
    return binario;
}

int binarioADecimal(int n) {
    int decimal = 0, base = 1;
    while (n > 0)
    {
        int ultimoDigito = n % 10;
        n /= 10;
        decimal += ultimoDigito * base;
        base *= 2;
    }
    return decimal;
}

int main() {
    int opcion, numero;
    // la entrada va a ser un string
    
    
    cout << "Elige un tipo de conversion:\n1. Decimal a Binario\n2. Binario a Decimal\n";
    cin >> opcion;

    if (opcion == 1) {
        cout << "Ingresa un numero decimal: ";
        cin >> numero;
        cout << "Binario: " << decimalABinario(numero) << endl;
    } else if (opcion == 2) {
        cout << "Ingrese un numero binario: ";
        cin >> numero;
        cout << "Decimal: " << binarioADecimal(numero) << endl;
    } else {
        cout << "Opcion invalida!" << endl;
    }

    return 0;
}